simpleline/event_loop/main_loop.py

```diff
@@ -92,7 +92,8 @@ def run(self):
         log.debug("Main loop ended. Running callback if set.")
 
         if self._quit_callback:
-            self._quit_callback()
+            cb = self._quit_callback.callback
+            cb(self._quit_callback.args)
```

Comment:
Callback's `args` could be a list of arguments and you could call `cb(*self._quit_callback.args)`. Then the callback can have more arguments than one.